### PR TITLE
Transformer bottleneck paper

### PR DESCRIPTION
**Pull request by isaacmg**
Merge everything

### CODE REVIEW

**Comment by isaacmg** [comment]
We should probably try to fix this error `To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
`
